apps/solver/equation_store.cpp

```diff
@@ -262,7 +271,9 @@ EquationStore::Error EquationStore::resolveLinearSystem(Expression exactSolution
   Preferences::AngleUnit angleUnit = Preferences::sharedPreferences()->angleUnit();
   // n unknown variables
   int n = 0;
-  while (m_variables[n][0] != 0) { n++; }
+  while (m_variables[n][0] != 0 && n < Expression::k_maxNumberOfVariables - 1) {
+    n++;
+  }
```

Comment:
Shouldn't it be:
'while (n < Expression::k_maxNumberOfVariables && m_variables[n][0] != 0) {
     n++;
   }'

Reply:
Indeed, thanks!